modules/core/src/main/java/org/apache/ignite/internal/table/criteria/SqlSerializer.java

```diff
@@ -232,5 +255,13 @@ public SqlSerializer build() {
 
             return ser;
         }
+
+        private static String normalizeIndexName(String name) {
+            if (!CANONICAL_NAME.matcher(name).matches()) {
+                throw new IllegalArgumentException("Index name must be alphanumeric with underscore and start with letter. Was: " + name);
+            }
```

Comment:
Let's 
1. split into validation and normalization functions.
2. move regex pattern and general validation method to some Utility class
3. the exception should be thrown here, in the current class, because here we have a context that we validate an index name.

Reply:
Done

Reply:
Another good candidate for such refactoring is `SystemView#normalizeIdentifier`